utils/utils.go

```diff
@@ -13,3 +13,10 @@ func CloseFile(file *os.File) {
 		log.Panicf("error during closing file %s. %s", err.Path, err.Error())
 	}
 }
+
+// Helper function to log errors
+func CheckError(err error, msg string) {
```

Comment:
Idea of this function is OK, earlier you had fatalF, which cause to exit after logging. Now since you have only printf, this error checking does not bring any value. You should properly handler error, Check if is present, if so do something, not just log it

Reply:
idk what to do here will think about it once I catch up on NM lecture